// @arco-design/web-vue/es/input/input-search.js
import { defineComponent, ref, createVNode, mergeProps, Fragment } from "vue";
import { getPrefixCls } from "../_utils/global-config.js";
import IconHover from "../_components/icon-hover.js";
import IconSearch from "../icon/icon-search/index.js";
import IconLoading from "../icon/icon-loading/index.js";
import Button from "../button/index.js";
import _Input from "./input.js";
import InputGroup from "./input-group.js";
import { SIZES } from "../_utils/constant.js";
var InputSearch = defineComponent({
  name: "InputSearch",
  inheritAttrs: false,
  props: {
    searchButton: {
      type: Boolean,
      default: false
    },
    loading: {
      type: Boolean,
      default: false
    },
    size: {
      type: String,
      default: "medium",
      validator: (value) => {
        return SIZES.includes(value);
      }
    },
    onSearch: {
      type: [Function, Array]
    }
  },
  emits: [
    "search"
  ],
  setup(props, {
    emit,
    slots,
    attrs
  }) {
    const prefixCls = getPrefixCls("input-search");
    const inputRef = ref();
    const handleClick = (e) => {
      if (inputRef.value.inputRef) {
        emit("search", inputRef.value.inputRef.value, e);
      }
    };
    const renderSuffix = () => {
      var _a;
      return createVNode(Fragment, null, [props.loading ? createVNode(IconLoading, null, null) : createVNode(IconHover, {
        "onClick": handleClick
      }, {
        default: () => [createVNode(IconSearch, null, null)]
      }), (_a = slots.suffix) == null ? void 0 : _a.call(slots)]);
    };
    const renderInput = () => {
      const inputSlots = {
        prepend: slots.prepend,
        prefix: slots.prefix,
        suffix: props.searchButton ? slots.suffix : renderSuffix,
        append: slots.append
      };
      return createVNode(_Input, mergeProps({
        "ref": inputRef,
        "size": props.size
      }, attrs), inputSlots);
    };
    const render = () => {
      if (props.searchButton) {
        return createVNode(InputGroup, null, {
          default: () => [renderInput(), createVNode(Button, {
            "type": "primary",
            "size": props.size,
            "class": `${prefixCls}-btn`,
            "loading": props.loading,
            "onClick": handleClick
          }, {
            icon: () => props.loading ? createVNode(IconLoading, null, null) : createVNode(IconSearch, null, null)
          })]
        });
      }
      return renderInput();
    };
    return {
      inputRef,
      render
    };
  },
  methods: {
    focus() {
      var _a;
      (_a = this.inputRef) == null ? void 0 : _a.focus();
    },
    blur() {
      var _a;
      (_a = this.inputRef) == null ? void 0 : _a.blur();
    }
  },
  render() {
    return this.render();
  }
});
export { InputSearch as default };
